examples/street-view-panorama-with-element/components/street-view-panorama-with-element/street-view-element.tsx

```diff
@@ -0,0 +1,31 @@
+import React, {useEffect, useRef} from 'react';
+
+import {
+  useGoogleMap,
+  useStreetViewPanorama
+} from '@ubilabs/google-maps-react-hooks';
+
+import styles from './street-view-element.module.css';
+
+const StreetViewPanoramaElement = () => {
+  const divRef = useRef<HTMLDivElement>(null);
+
+  const map = useGoogleMap();
+
+  const panorama = useStreetViewPanorama({
+    divElement: divRef.current,
+    position: {lat: 53.55150164023877, lng: 9.986843327204179},
+    pov: {heading: 165, pitch: 0},
+    zoom: 1
+  });
+
+  useEffect(() => {
+    if (map) {
+      map.setStreetView(panorama);
+    }
+  }, [map]);
```

Comment:
```suggestion
  }, [map, panorama]);

  // Clean up map when component unmounts
  useEffect(
    () => () => {
      if (map) {
        map.setStreetView(null);
      }
    },
    []
  );
```

Reply:
And then also:

`if (map && panorama) {}` for line 23 and line 31, right?

Why a new `useEffect` hook and not the cleanup above in the hook with the dependencies?

Reply:
No, `if (map)` is good how it is. If the panorama changes to `null`, we also want to set the street view instance of the map to `null`.

If we do the clean up in the hook with the dependencies and the panorama instance changes, this would call
```js
map.setStreetView(panoramaInstance1);
map.setStreetView(null); // because of the clean up
map.setStreetView(panoramaInstance2);
```
but we would like to have
```js
map.setStreetView(panoramaInstance1);
map.setStreetView(panoramaInstance2);
```
and only want to reset the map's street view to `null` when the component unmounts. So we need two `useEffect`s here, one with the dependencies to update the instance and one without dependencies to run only on mount/unmount.